rfcs/2024-03-22-20170-trace-data-model.md

```diff
@@ -0,0 +1,196 @@
+# RFC 20170 - 2024-03-22 - Establish an Internal Trace Model
+
+As of today there is an already existing Log and Metric event model, the purpose is to establish an
+internal model for describing a trace event.
+
+## Context
+
+The [Ingest OpenTelemetry Traces RFC] was accepted, but there was a condition that an [internal trace model] MUST be
+established. Meanwhile, there's also the [Accept Datadog Traces RFC] which was established, but
+that primarily only works well with the [`datadog_traces` sink]. There was also a previous RFC which
+was concerned with [validating schemas], but never came attempted to define an event schema.  This leaves this RFC
+to establish that data model.
+
+## Cross cutting concerns
+
+- This is part of the effort to support [sending OpenTelemetry traces], it would also support
+  ingesting OpenTelemetry. As well as to provide a path for a canonical model which traces can be
+  translated to/from ideally without loss of fidelity.
```

Comment:
```suggestion
- This is part of the effort to support [sending OpenTelemetry traces] as well as ingesting them.
   This requires a canonical trace data model which can carry these traces without loss of fidelity.
```

Reply:
Note, that this is also not a cross-cutting concern, as it is one of the _core_ issues for developing this model.